nexus/db-queries/src/db/datastore/saga.rs

```diff
@@ -99,4 +104,277 @@ impl DataStore {
             )),
         }
     }
+
+    /// Returns a list of unfinished sagas assigned to SEC `sec_id`, making as
+    /// many queries as needed (in batches) to get them all
+    pub async fn saga_list_recovery_candidates_batched(
+        &self,
+        opctx: &OpContext,
+        sec_id: &db::saga_types::SecId,
+    ) -> Result<Vec<db::saga_types::Saga>, Error> {
+        let mut sagas = vec![];
+        let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+        let conn = self.pool_connection_authorized(opctx).await?;
+        while let Some(p) = paginator.next() {
+            use db::schema::saga::dsl;
+
+            let mut batch =
+                paginated(dsl::saga, dsl::id, &p.current_pagparams())
+                    .filter(dsl::saga_state.ne(
+                        db::saga_types::SagaCachedState(
+                            steno::SagaCachedState::Done,
+                        ),
+                    ))
+                    .filter(dsl::current_sec.eq(*sec_id))
+                    .select(db::saga_types::Saga::as_select())
+                    .load_async(&*conn)
+                    .await
+                    .map_err(|e| {
+                        public_error_from_diesel(e, ErrorHandler::Server)
+                    })?;
+
+            paginator = p.found_batch(&batch, &|row| row.id);
+            sagas.append(&mut batch);
+        }
+        Ok(sagas)
+    }
+
+    /// Returns a list of all saga log entries for the given saga, making as
+    /// many queries as needed (in batches) to get them all
+    pub async fn saga_fetch_log_batched(
+        &self,
+        opctx: &OpContext,
+        saga: &db::saga_types::Saga,
```

Comment:
I think I preserved what the function was doing, and I had the same thought, but I opted to leave it in case we do wind up wanting some of that state in the future.  That was more relevant in development than now that it's ready to land though.

Reply:
fair enough, carry on!